src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1146,10 +1147,12 @@ protected override void GetItem(string path)
                         }
                     }
                 }
+#endif
 
                 FileSystemInfo result = GetFileSystemItem(path, ref isContainer, false);
                 if (result != null)
                 {
+#if !UNIX
```

Comment:
Actually, if you don't mind duplicating the WriteItemObject line, it can be a pure #if #else block.  Your call.